sdk/core/System.ClientModel/src/Options/DiagnosticsOptions.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.ClientModel.Options;
+
+/// <summary>
+/// TODO
+/// </summary>
+public class DiagnosticsOptions
+{
+    private bool _frozen;
+
+    private bool _isLoggingEnabled = true;
+    private int _loggedContentSizeLimit = 4 * 1024;
+    private bool _isLoggingContentEnabled;
+    private IEnumerable<string>? _loggedHeaderNames;
+    private IEnumerable<string>? _loggedQueryParameters;
+    private string? _clientAssembly;
+    private string? _requestIdHeaderName;
+
+    /// <summary>
+    /// Get or sets value indicating whether HTTP pipeline logging is enabled.
+    /// </summary>
+    public bool IsLoggingEnabled
+    {
+        get => _isLoggingEnabled;
+        set
+        {
+            AssertNotFrozen();
+
+            _isLoggingEnabled = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if request or response content should be logged.
+    /// </summary>
+    public bool IsLoggingContentEnabled
+    {
+        get => _isLoggingContentEnabled;
+        set
+        {
+            AssertNotFrozen();
+
+            _isLoggingContentEnabled = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating maximum size of content to log in bytes. Defaults to 4096.
+    /// </summary>
+    public int LoggedContentSizeLimit
+    {
+        get => _loggedContentSizeLimit;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggedContentSizeLimit = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a list of header names that are not redacted during logging.
+    /// </summary>
+    public IEnumerable<string>? LoggedHeaderNames
+    {
+        get => _loggedHeaderNames;
+        set
```

Comment:
The actual IEnumerables are not freezable. They are copied into arrays before creating the sanitizer when the pipeline is created - is this enough?